azurerm/resource_arm_network_interface_test.go

```diff
@@ -333,6 +333,32 @@ func TestAccAzureRMNetworkInterface_withTags(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMNetworkInterface_IPAddressesBug1286(t *testing.T) {
+	resourceName := "azurerm_network_interface.test"
+	rInt := acctest.RandInt()
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMNetworkInterfaceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMNetworkInterface_withIPAddresses(rInt, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMNetworkInterfaceExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "ip_configuration.0.private_ip_address"),
+					resource.TestCheckResourceAttrSet(resourceName, "ip_configuration.0.public_ip_address_id"),
+				),
+			},
+			{
+				Config: testAccAzureRMNetworkInterface_withIPAddressesUpdate(rInt, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMNetworkInterfaceExists(resourceName),
```

Comment:
we can also verify these are set to nothing (e.g. there's no Public/Private IP Addresses set) via:

```
resource.TestCheckResourceAttr(resourceName, "ip_configuration.0.private_ip_address", ""),
resource.TestCheckResourceAttr(resourceName, "ip_configuration.0. public_ip_address_id", ""),
```